src/overlays/actors/ovl_En_Dekubaba/z_en_dekubaba.c

```diff
@@ -44,188 +54,1324 @@ const ActorInit En_Dekubaba_InitVars = {
+    for (i = 0; i < stemSections; i++) {
+        mf.yw += 20.0f * Math_SinS(this->stemSectionAngle[i]) * this->size;
+        horizontalStepSize = Math_CosS(this->stemSectionAngle[i]) * 20.0f * this->size;
+        mf.xw -= horizontalStepSize * Math_SinS(this->actor.shape.rot.y);
+        mf.zw -= horizontalStepSize * Math_CosS(this->actor.shape.rot.y);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B2C40.s")
+        Matrix_Put(&mf);
+        Matrix_RotateZYX(this->stemSectionAngle[i], this->actor.shape.rot.y, 0, MTXMODE_APPLY);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, sStemDLists[i]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B2CB8.s")
+        Collider_UpdateSpheres(51 + 2 * i, &this->collider);
+        Collider_UpdateSpheres(52 + 2 * i, &this->collider);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B3044.s")
+        if (i == 0) {
+            if (this->actionFunc != EnDekubaba_Sway) {
+                this->actor.focus.pos.x = mf.xw;
+                this->actor.focus.pos.y = mf.yw;
+                this->actor.focus.pos.z = mf.zw;
+            } else {
+                this->actor.focus.pos.x = this->actor.home.pos.x;
+                this->actor.focus.pos.y = this->actor.home.pos.y + (40.0f * this->size);
+                this->actor.focus.pos.z = this->actor.home.pos.z;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B3078.s")
+        Matrix_MultZero(&this->bodyPartsPos[i + 1]);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B3170.s")
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B3280.s")
+void EnDekubaba_DrawStemBasePruned(EnDekubaba* this, PlayState* play) {
+    OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B3390.s")
+    Matrix_RotateZYX(this->stemSectionAngle[2], this->actor.shape.rot.y, 0, MTXMODE_APPLY);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, gDekuBabaStemBaseDL);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B3404.s")
+    Collider_UpdateSpheres(55, &this->collider);
+    Collider_UpdateSpheres(56, &this->collider);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B3768.s")
+    Matrix_MultZero(&this->bodyPartsPos[3]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B37E8.s")
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dekubaba/func_808B39AC.s")
+void EnDekubaba_DrawBaseShadow(EnDekubaba* this, PlayState* play) {
```

Comment:
```suggestion
void EnDekubaba_DrawShadow(EnDekubaba* this, PlayState* play) {
```
Simpler name since the actor only has one shadow.

Reply:
Probably the point here was that only the base has the shadow, there's isn't one for the rest, but I can change if preferred.

Reply:
I prefer the shorter version, but if you have a heavy preference to keep as-is that's fine